File: examples/undocumented/r_modular/kernel_distance_modular.R
library(shogun)

fm_train_real <- t(as.matrix(read.table('../data/fm_train_real.dat')))
fm_test_real <- t(as.matrix(read.table('../data/fm_test_real.dat')))

# distance
print('Distance')

feats_train <- RealFeatures()
dummy <- feats_train$set_feature_matrix(fm_train_real)
feats_test <- RealFeatures()
dummy <- feats_test$set_feature_matrix(fm_test_real)
width <- 1.7
distance <- EuclideanDistance()

kernel <- DistanceKernel(feats_train, feats_test, width, distance)

km_train <- kernel$get_kernel_matrix()
dump <- kernel$init(feats_train, feats_test)
km_test <- kernel$get_kernel_matrix()

File: examples/undocumented/csharp_modular/kernel_distance_modular.cs
using System;

public class kernel_distance_modular {
	public static void Main() {
		modshogun.init_shogun_with_defaults();
		double width = 1.7;

		double[,] traindata_real = Load.load_numbers("../data/fm_train_real.dat");
		double[,] testdata_real = Load.load_numbers("../data/fm_test_real.dat");

		RealFeatures feats_train = new RealFeatures(traindata_real);
		RealFeatures feats_test = new RealFeatures(testdata_real);

		EuclideanDistance distance = new EuclideanDistance();

		DistanceKernel kernel = new DistanceKernel(feats_train, feats_test, width, distance);

		double[,] dm_train = distance.get_distance_matrix();
		distance.init(feats_train, feats_test);
		double[,] dm_test = distance.get_distance_matrix();

		//  Parse and Display km_train
		Console.Write("dm_train:\n");
		int numRows = dm_train.GetLength(0);
		int numCols = dm_train.GetLength(1);

		for(int i = 0; i < numRows; i++){
			for(int j = 0; j < numCols; j++){
				Console.Write(dm_train[i,j] +" ");
			}
			Console.Write("\n");
		}

		//  Parse and Display km_test
		Console.Write("\ndm_test:\n");
		numRows = dm_test.GetLength(0);
		numCols = dm_test.GetLength(1);

		for(int i = 0; i < numRows; i++){
			for(int j = 0; j < numCols; j++){
				Console.Write(dm_test[i,j] +" ");
			}
			Console.Write("\n");
		}


		modshogun.exit_shogun();
	}
}
